tests/unit_tests/sparse_blas/include/test_common.hpp

```diff
@@ -113,6 +127,28 @@ void rand_vector(std::vector<fpType> &v, std::size_t n) {
     }
 }
 
+template <typename fpType>
+void rand_matrix(std::vector<fpType> &m, oneapi::mkl::layout layout_val, std::size_t nrows,
```

Comment:
It seems a bit pointless to m as an in-out parameter instead of just creating the vector in the function.

Reply:
I agree but this is more consistent with other domains, blas has very similar functions bu that code is not shared across domains. I have a preference to keep it as is.